types/part_set.go

```diff
@@ -186,6 +189,7 @@ func NewPartSetFromData(data []byte, partSize uint32) *PartSet {
 		parts:         parts,
 		partsBitArray: partsBitArray,
 		count:         total,
+		byteSize:      int64(len(data)),
```

Comment:
i think this is culprit, you should be accounting for the parts not the entire struct